icu4j/main/core/src/main/java/com/ibm/icu/text/RuleBasedNumberFormat.java

```diff
@@ -1902,40 +1870,34 @@ private StringBuilder stripWhitespace(String description) {
 
         // iterate through the characters...
         int start = 0;
+        char ch;
         while (start < descriptionLength) {
-            // seek to the first non-whitespace character...
+            // Seek to the first non-whitespace character...
+            // If the first non-whitespace character is semicolon, skip it and continue
             while (start < descriptionLength
-                   && PatternProps.isWhiteSpace(description.charAt(start)))
+                   && (PatternProps.isWhiteSpace(ch = description.charAt(start)) || ch == ';'))
             {
                 ++start;
             }
 
-            //if the first non-whitespace character is semicolon, skip it and continue
-            if (start < descriptionLength && description.charAt(start) == ';') {
-                start += 1;
-                continue;
-            }
-
```

Comment:
Some of this logic in stripWhitespace was out of sync between C++ and Java. Both sides have been brought in sync in this pull request.